### DISCUSSION TITLE
dietpi-backup - encrypted backup support

### DISCUSSION THREAD

**Discussion by mstgrv** [MichaIng/DietPi#6344] [Software Request]
Would be great if dietpi-backup supported encrypted backups - that is, backups created by the tool are encrypted at rest on the destination, and need to be decrypted before they can be restored anywhere.

**Comment by MichaIng** [comment]
It doubles with the request here, right? #6081

**Comment by mstgrv** [comment]
There's some overlap, but backing up to a remote target isn't part of what I'm trying to do - just make encrypted backups to direct attached storage. Compressing backups isn't what I was thinking either, but if that's a way to achieve encryption then I guess that overlaps as well.

**Reply by MichaIng** [reply]
Yeah, when I find time, I'd implement it in a combined way, for local and remote targets, as optionally encrypted compressed archive. If this extra processing is required anyway, I see only upsides if it is compressed in the same turn.

**Reply by mstgrv** [reply]
Sounds good! If it's an easier first step to do archived & encrypted backups to local targets first, that'd be great also

**Reply by MichaIng** [reply]
We do not handle local and remote targets differently. It's both syncing to a directory, whether local or NFS or SSHFS mount. I was thinking about using `rsync`'s direct remote SSH or `rsync` daemon feature, but do not see so much benefit (as it has limitations as well): #2364

**Reply by mstgrv** [reply]
Ah cool thanks, didn't realise remote targets were already supported (thought that was also a feature request)
